### PR TITLE
Allow Renaming Teams

### LINKED ISSUE #5067: Allow users to rename teams
### Problem to solve

The upcoming changes for Teams & Projects do not include a way to rename a team.

### Proposal

In the long run, having team settings could be interesting and also a good place to surface a team rename action. However, a minimum viable change (MVC) for this could be to a) add a more actions button next to the primary action button on the team page (projects list) and b) trigger a confirmation modal to rename a team. See also closely related issue about deleting teams in https://github.com/gitpod-io/gitpod/issues/5066 for specs.

### PR DESCRIPTION
**Pull request by svenefftinge**
Allow updating the team's name

fixes #5067

## How to test

rename team name

## Release Notes

```release-note
Allow renaming teams
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by gtsiolis** [comment]
Looking at this now! 👀

File: components/dashboard/src/teams/TeamSettings.tsx [hunk 1]
```diff
@@ -60,20 +62,57 @@ export default function TeamSettings() {
...
+
+    const onNameChange = useCallback(
+        async (event: React.ChangeEvent<HTMLInputElement>) => {
+            if (!team) {
+                return;
+            }
+            const newName = event.target.value || "";
+            setTeamName(newName);
+            if (newName.trim().length === 0) {
+                setErrorMessage("Team name must not be empty.");
```
Comment (gtsiolis) [inline components/dashboard/src/teams/TeamSettings.tsx:91]: **suggestion**: We should probably update the saving and feedback interaction to also match the UX on the user profile, but this is out of the scope of this PR. 🍒

File: components/dashboard/src/teams/TeamSettings.tsx [hunk 2]
```diff
@@ -82,7 +121,39 @@ export default function TeamSettings() {
...
+                    </Alert>
+                )}
+                {updated && (
+                    <Alert type="message" closable={true} className="mb-2 max-w-xl rounded-md">
+                        Profile information has been updated.
+                    </Alert>
+                )}
+                <div className="flex flex-col lg:flex-row">
+                    <div>
+                        <div className="mt-4">
```
Comment (gtsiolis) [inline components/dashboard/src/teams/TeamSettings.tsx:140]: **suggestion**: The amount of space between items creates relationships and hierarchy.

```suggestion
                        <div className="mt-4 mb-3">
```
---

Posting below a screenshot from **Foundations / Spacing** in the component library in our Figma team.

| Spacing Guidelines |
|-|
| Spacing |

File: components/dashboard/src/teams/TeamSettings.tsx [hunk 3]
```diff
@@ -82,7 +121,39 @@ export default function TeamSettings() {
...
+                            <h4>Name</h4>
+                            <input type="text" value={teamName} onChange={onNameChange} />
+                        </div>
+                    </div>
+                </div>
+                <div className="flex flex-row mt-8">
```
Comment (gtsiolis) [inline components/dashboard/src/teams/TeamSettings.tsx:146]: **suggestion**: Adding the spacing on the form element above should eliminate the need for the extra top margin on the button.

```suggestion
                <div className="flex flex-row">
```

**Review by easyCZ — APPROVED** [review]
LGTM % modulo better signature for the update funcation

/hold

File: components/dashboard/src/teams/TeamSettings.tsx [hunk 4]
```diff
@@ -82,7 +121,39 @@ export default function TeamSettings() {
...
+                    This is your team's visible name within Gitpod. For example, the name of your company.
+                </p>
+                {errorMessage && (
+                    <Alert type="error" closable={true} className="mb-2 max-w-xl rounded-md">
+                        {errorMessage}
+                    </Alert>
+                )}
+                {updated && (
+                    <Alert type="message" closable={true} className="mb-2 max-w-xl rounded-md">
+                        Profile information has been updated.
```
Comment (gtsiolis) [inline components/dashboard/src/teams/TeamSettings.tsx:135]: **issue**: This was still referring to the user profile information-section.

```suggestion
                        Team name has been updated.
```

File: components/dashboard/src/teams/TeamSettings.tsx [hunk 5]
```diff
@@ -82,7 +121,39 @@ export default function TeamSettings() {
...
+                        </div>
+                    </div>
+                </div>
+                <div className="flex flex-row mt-8">
+                    <button
+                        className="primary"
+                        disabled={team?.name === teamName || !!errorMessage}
+                        onClick={updateTeamInformation}
+                    >
+                        Update Name
```
Comment (gtsiolis) [inline components/dashboard/src/teams/TeamSettings.tsx:152]: **though**:: I guess this is known, but this is not changing the team slug. I'd expect to get some user feedback on this to add a section for updating the team URL. This is also out of the scope of this PR.

**Review by gtsiolis — APPROVED** [review]
Thanks for making this change, @svenefftinge! 🍷 

UX LGTM!

Left a few non-blocking comments that can be moved to follow-up issues.

**Comment by easyCZ** [comment]
Should this be dependant on https://github.com/gitpod-io/gitpod/pull/15749? What's the relationship to that PR? They seem to share the changes

File: components/server/src/workspace/gitpod-server-impl.ts [hunk 6]
```diff
@@ -2061,6 +2061,24 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
         return team;
     }
 
+    public async updateTeam(ctx: TraceContext, teamId: string, team: Partial<Pick<Team, "name">>): Promise<Team> {
```
[the remark this answers is not in the document]
Comment (svenefftinge) [reply components/server/src/workspace/gitpod-server-impl.ts:2064]: We are soon going to add additional properties, such as avatar. I didn't want to change the API then.

**Comment by svenefftinge** [comment]
> Should this be dependant on https://github.com/gitpod-io/gitpod/pull/15749? What's the relationship to that PR? They seem to share the changes

I based the UI work on that branch.  I can merge just this one as well.

File: components/dashboard/src/teams/TeamSettings.tsx (hunk 5 above)
Comment (svenefftinge) [reply components/dashboard/src/teams/TeamSettings.tsx:152]: I'd liek to get rid of the team URL tbh

File: components/dashboard/src/teams/TeamSettings.tsx [hunk 7]
```diff
@@ -82,7 +122,39 @@ export default function TeamSettings() {
...
+                        <div className="mt-4 mb-3">
+                            <h4>Name</h4>
+                            <input type="text" value={teamName} onChange={onNameChange} />
+                        </div>
+                    </div>
+                </div>
+                <div className="flex flex-row">
+                    <button
+                        className="primary"
+                        disabled={team?.name === teamName || !!errorMessage}
```
Comment (filiptronicek) [inline components/dashboard/src/teams/TeamSettings.tsx:150]: nit: currently, the user is able to change the team name to the same name just with whitespaces around it.

```suggestion
                        disabled={team?.name === teamName.trim() || !!errorMessage}
```

**Pull request merged**